drud/ddev
# Make mysql client history survive a ddev restart

## The Problem/Issue/Bug:
It would be nice if the SQL statements written to the client history file were still there after a `ddev restart`.
Currently they are not.

## How this PR Solves The Problem:
It sets an environment variable so the mysql client will write
its history to a location on a persistent volume.
That location is shared between the web and the db container.

## Manual Testing Instructions:

- Remove the existing history files with `ddev start && ddev exec rm -rf /mnt/ddev-global-cache/mysqlhistory`
- `ddev restart` and verify that the directories now exist, `ddev exec ls -lR /mnt/ddev-global-cache/mysqlhistory`
- `ddev drush sqlc`
- Issue an SQL statement
- Quit
- `ddev restart`
- `ddev drush sqlc`
- Hit arrow up -> Statement should still be there, in v1.17.7 it is not.

## Notes:
1. There is no variable to limit the size of the file like there is for bash. From https://planet.mysql.com/entry/?id=5991652 : 
```
Despite the similarities to Bash, there is no MYSQL_HISTSIZE environment variable
to be like Bash's HISTSIZE. With some unreliable fiddling, 
one can try to tell the underlying library what the maximum history size should be
[...]
Usually this particular trick is best avoided because of its side effects.
```
2. The directory is not created automatically, if it doesn't already exist the history file is not written.

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3113"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
